### DISCUSSION TITLE
Insert image in choices?

### DISCUSSION THREAD

**Discussion by ler93** [jspsych/jsPsych#2200] [Q&A] [answered]
Hi! Part of my experiment consists of applying a short version of the raven test. I'm using the "image-button-response" plugin to place the images. What I did is add in "choices" a numeric value: 1, 2, 3 etc. for participants to choose the option they consider correct. I was wondering if there is any way to replace the numbers with the images of the different options so that the subjects touch directly on the image and not on the number that corresponds to the option. I hope I was clear! Thanks

**Comment by becky-gilbert** [comment, marked as answer]
Hi @ler93, yes you can do that. You can set up the `choices` array so that it contains the image file paths. Then, use the `button_html` parameter to add an `img` element into the button, which should have a `src` attribute whose value is the file that comes from the `choices` array.  The second example on the [documentation page for the `audio-button-response` plugin](https://www.jspsych.org/7.0/plugins/audio-button-response/#examples) shows how to do this. 

I hope that helps - let us know if you need any further guidance.

**Reply by ler93** [reply]
Thanks Becky! Your answer helped me a lot.

**Comment by ler93** [comment]
Hello again, @becky-gilbert . Sorry to reopen this conversation. I tried the method you indicated but the problem is that when I position over any of the choices (now images), the cursor doesn't change from the "arrow" to the "hand". I'm afraid that participants may doubt about the place where they should click. Is there any way to change this? Just in case, I copy the code below.
(I'm working on cognition.run, maybe it could be related to that?)

```
var timeline = [];

     var image_raven1 = {
       type: 'image-button-response',
       stimulus: 'short_raven1.png',
       choices: ['short_raven1_option1.png', 'short_raven1_option2.png', 'short_raven1_option3.png', 'short_raven1_option4.png', 'short_raven1_option5.png', 'short_raven1_option6.png'],
       button_html: '<img src = "% choice%" />',
     };
     timeline.push (image_raven1);

jsPsych.init ({
timeline: timeline,
});
```

**Reply by becky-gilbert** [reply]
Hmm, that's odd - I think there might be an error in the example code that I linked to. Try this:
```js
button_html: '<button class="jspsych-btn"><img src = "%choice%" /></button>'
```

**Reply by ler93** [reply]
It worked perfectly! Thank you very much, Becky!

**Comment by ler93** [comment]
Hi again, @becky-gilbert. Is there a way for choices to appear on two different lines, one below the other? (I would like the raven test options to appear as in the original test: 4 above and 4 below). Thanks and sorry for so many concerns! 

The code for this line is: 

'''


     var image_raven1 = {
       type: 'image-button-response',
       stimulus: 'example1.png',
       choices: ['example1_option1.png', 'example1_option2.png', 'example1_option3.png', 'example1_option4.png',
       'example1_option5.png', 'example1_option6.png', 'example1_option7.png', 'example1_option8.png'],
       button_html: '<button class = "jspsych-btn"> <img src = "% choice%" /> </button>',
     };
     timeline.push (image_raven1);

'''

**Reply by becky-gilbert** [reply]
Hi @ler93, this is something you'll need to do using CSS. There are lots of ways to change the button position layout. I think one way to do this is by reducing the width of the div that holds all of the buttons/images to 50% of it's normal width, so that only half of the buttons/images will fit horizontally and the rest of the buttons/images will have to wrap down into a new row. If you do this, then you'll also need to re-center div container (because reducing it's width to 50% will make it appear on the left-side of the screen).

The ID of the button container in the `image-buttton-response` plugin is "jspsych-image-button-response-btngroup", so you'll want to create a CSS selector for this ID. Then, the CSS rules you're adding for this element are (1) `width: 50%` and (2) `transform: translate(50%, 0)` (to re-center the div along the x-axis). You can put these CSS rules into a `<style>` section in the `<head>` section of your HTML file:
```html
<head>
  /* load the JavaScript files, etc */
  <script src="..."></script>
  /* add a style section here, if you don't have one already */
  <style>
    #jspsych-image-button-response-btngroup {
      width: 50%;
      transform: translate(50%,0);
    }
  </style>
</head>
<body></body>
<script>
  // jsPsych experiment code ...
```
Does that solution work for you?

Also, you might find the jsPsych [Controlling Visual Appearance documentation page](https://www.jspsych.org/7.0/overview/style/) useful for CSS stuff. The section [Tips for working with CSS](https://www.jspsych.org/7.0/overview/style/#tips-for-working-with-css) might be especially useful as it explains how to use the developer tools to identify the elements on the page and immediately see the results of changing their CSS rules.

**Reply by ler93** [reply]
Becky! I was able to do it with the code you send me. I wanted to comment what steps I followed, in case someone working in cognition.run face the same problem in the future.
The part of the code that I needed to save in visual studio code was:

```
#jspsych-image-button-response-btngroup {
       width: 50%;
       transform: translate (50%, 0);
     }
```
Done this, save the file in CSS format, and I just uploaded it to cognition.run as external JS/CSS. The changes were automatically applied to my experiment.

**Reply by becky-gilbert** [reply]
Great, thanks so much for the update, and for clarifying the steps!
